pkg/apis/navigator/v1alpha1/defaults.go

```diff
@@ -7,3 +7,13 @@ import (
 func addDefaultingFuncs(scheme *runtime.Scheme) error {
 	return RegisterDefaults(scheme)
 }
+
+func SetDefaults_CassandraClusterSpec(spec *CassandraClusterSpec) {
```

Comment:
Why default the cassandraclusterspec if we're only going to then iterate over each node pool in it anyway?

Surely this is better to be a `SetDefaults_CassandraClusterNodePool` function?

This leads onto my next question.. what if a user wants to ensure there are 0 seeds in a given node pool (as they have seeds in other node pools). Is this something we want to disallow?

Reply:
Yep, it'd be cleaner with `SetDefaults_CassandraClusterNodePool`.

We need to ensure that there's more than 0 seeds across all nodepools. We could reduce the minimum to 0 in a particular nodepool if we find a use-case for it (I can't find a reason to do it)

Reply:
Cool - I've no issue requiring minimum one seed per node pool for now. We can revisit it in future if needed like you say.

Reply:
I've opened https://github.com/jetstack/navigator/issues/272 for posterity